File: src/main/java/me/alegian/thavma/impl/client/gui/tooltip/TooltipHelper.java
package me.alegian.thavma.impl.client.gui.tooltip;

import me.alegian.thavma.impl.common.aspect.AspectMap;
import me.alegian.thavma.impl.init.registries.deferred.Aspects;
import net.minecraft.network.chat.Component;
import net.minecraft.network.chat.MutableComponent;

public class TooltipHelper {
  public static MutableComponent containedPrimals(AspectMap contents) {
    var sequence = Component.empty();
    for (int i = 0; i < Aspects.INSTANCE.getPRIMAL_ASPECTS().size(); i++) {
      var a = Aspects.INSTANCE.getPRIMAL_ASPECTS().get(i).get();
      var newPart = Component.literal(String.valueOf(contents.get(a))).withColor(a.getColor());
      sequence.append(newPart);
      if (i != Aspects.INSTANCE.getPRIMAL_ASPECTS().size() - 1) sequence.append(Component.literal(" | "));
    }
    return sequence;
  }
}

File: src/main/java/me/alegian/thavma/impl/client/gui/tooltip/TooltipHelper.kt
package me.alegian.thavma.impl.client.gui.tooltip

import me.alegian.thavma.impl.common.aspect.AspectMap
import me.alegian.thavma.impl.init.registries.deferred.Aspects.PRIMAL_ASPECTS
import net.minecraft.network.chat.Component
import net.minecraft.network.chat.MutableComponent

fun containedPrimalsComponent(contents: AspectMap): MutableComponent {
    return Component.empty().also {
        for (i in PRIMAL_ASPECTS.indices) {
            val a = PRIMAL_ASPECTS[i].get()
            val newPart = Component.literal(contents[a].toString()).withColor(a.color)
            it.append(newPart)
            if (i != PRIMAL_ASPECTS.size - 1) it.append(Component.literal(" | "))
        }
    }
}
